src/python/pants/backend/jvm/tasks/scalafix.py

```diff
@@ -51,19 +52,14 @@ def prepare(cls, options, round_manager):
     if options.semantic:
       round_manager.require_data('runtime_classpath')
 
-  def _compute_classpath(self, targets):
-    classpaths = self.context.products.get_data('runtime_classpath')
-    return [entry for _, entry in classpaths.get_for_targets(targets)]
-
   def invoke_tool(self, absolute_root, target_sources):
     args = []
     tool_classpath = self.tool_classpath('scalafix-tool-classpath')
     if tool_classpath:
       args.append('--tool-classpath={}'.format(os.pathsep.join(tool_classpath)))
     if self.get_options().semantic:
-      # If semantic checks are enabled, pass the relevant classpath entries for these
-      # targets.
-      classpath = self._compute_classpath({target for target, _ in target_sources})
+      # If semantic checks are enabled, we need the full classpath for these targets.
+      classpath = self.classpath({target for target, _ in target_sources})
```

Comment:
is this doing the right thing when you test? looking at the test failures it looks like that constructed classpath isn't correct, see:
`No default jvm platform is defined` in https://travis-ci.org/pantsbuild/pants/jobs/556896821#L430 and similar errors in other shard failures.

I would suggest running that ^ test locally and inspecting the classpath generated.

Reply:
this `classpath = ...` line should only be be hit when we run `scalafix` with `--semantic`, not sure if that happens anywhere.